File: Backend/app/business/models/estudiante.py
from app import db
from datetime import datetime

class Estudiante(db.Model):
    __tablename__ = 'estudiantes'

    id         = db.Column(db.Integer, primary_key=True)
    persona_id = db.Column(db.Integer, db.ForeignKey('personas.id'), nullable=False)
    institucion = db.Column(db.String(150), nullable=True)
    carrera    = db.Column(db.String(100), nullable=True)
    horario    = db.Column(db.JSON, nullable=True)
    # Ejemplo de horario:
    # {
    #   "lunes":    ["8:00 Cálculo", "10:00 Inglés"],
    #   "martes":   ["9:00 Programación"],
    #   "miercoles":[]
    # }

    # Relaciones
    persona  = db.relationship('Persona', back_populates='estudiante')
    sesiones = db.relationship('Sesion', back_populates='estudiante',
                               cascade='all, delete-orphan')

    def __repr__(self):
        return f'<Estudiante {self.id}>'

    def to_dict(self):
        return {
            'id':          self.id,
            'persona_id':  self.persona_id,
            'institucion': self.institucion,
            'carrera':     self.carrera,
            'horario':     self.horario,
            'persona':     self.persona.to_dict() if self.persona else None
        }

File: Frontend/src/app/models/Estudiante.ts
import { Persona } from './Persona';

export interface HorarioItem {
  titulo: string;
  hora: string;
  tipo?: 'clase' | 'estudio' | 'descanso' | 'otro';
}

export interface Horario {
  lunes?: HorarioItem[];
  martes?: HorarioItem[];
  miercoles?: HorarioItem[];
  jueves?: HorarioItem[];
  viernes?: HorarioItem[];
  sabado?: HorarioItem[];
  domingo?: HorarioItem[];
}

export class Estudiante {
  id?: number;
  persona_id?: number;
  institucion?: string;
  carrera?: string;
  horario?: Horario;
  persona?: Persona;
}
